protocol.go

```
@@ -583,6 +602,9 @@ func (c *rawConnection) writerLoop() {
 			if hm.msg != nil {
 				// Uncompressed message in uncBuf
 				uncBuf, err = hm.msg.AppendXDR(uncBuf[:0])
+				if hm.done != nil {
+					hm.done <- struct{}{}
```

Generally speaking I prefer `close(someChannel)` over `someChannel <- struct{}{}` for done signals, as the close can't block, can notify multiple listeners, and panics instead of blocking forever if there's a logic error and we try to be done more than once...